src/main/typescript/node_modules/@atomist/rug/test/handler/WellKnownSteps.ts

```diff
@@ -2,18 +2,35 @@ import { Given, Then, When } from "./Core";
 
 // Register well-known steps
 
+/**
+ * Nothing.  Typical starting point for command handler testing.
+ */
+Given("nothing", (w) => { return; });
+
+/**
+ * Generic event handler registrtion.
+ */
 Given("([a-zA-Z0-9]+) handler", (world, handlerName) =>
     world.registerHandler(handlerName),
 );
 
+/**
+ * No event handler was triggered.
+ */
 Then("no handler fired", (world) =>
     world.plan() === null,
 );
 
+/**
+ * Invalid command handler parameters.
+ */
 Then("handler parameters were invalid", (world) =>
     world.invalidParameters() != null,
 );
 
```

Comment:
Do you think it would be helpful? After thinking about it I figured the Plan + onError would be sufficient for handlers to communicate failure. Project Rugs do not have a similar mechanism to communicate back to their caller, so throwing exceptions is the only option.